pkg/operator/ceph/agent/agent.go

```diff
@@ -240,6 +242,35 @@ func (a *Agent) createAgentDaemonSet(namespace, agentImage, serviceAccount strin
 		}
 	}
 
+	// Add NodeAffinity if any
+	nodeLabelKey := os.Getenv(agentDaemonsetNodeLabelKeyEnv)
+	nodeLabelValue := os.Getenv(agentDaemonsetNodeLabelValueEnv)
+
+	var nodeLabelValues []string
```

Comment:
Could you create a helper method for adding the node affinity? For example, it could be added to pkg/operator/k8sutil/node.go. Then both the agent and the discovery could both call it.

Reply:
Sounds good. Will work on it.